### PR TITLE
Opets/us129851 truncation message for quiz details page

### PR DESCRIPTION
**Pull request by opets**
Quality Notes
- Functional Testing
- [x]   Browsers (Chrome, FF, Edge)
- [x]   Accessibility
- [x]   RTL
- [x]   Responsive
- [x]   Test cases
- Security, devops, perf: N/A
- UX and docs
- [x]   Demo
FYI @Brightspace/draco-vis

### CODE REVIEW

File: locales/en.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ export default {
 	"dashboard:item": "item",
 	"dashboard:items": "items",
 	"dashboard:orgUnitFilterName": "Org Unit",
+	"dashboard:quizTruncationTitle": "Your results exceed {num} quizzes and we are unable to display all available information.",
+	"dashboard:quizTruncationSubtext": "Please refine your filters for contact D2L Support for assistance.",
```
Comment (bemailloux) [inline locales/en.js:44]: I think this should be "~~for~~ *or* contact D2L Support"

**Review by bemailloux — APPROVED** [review]
lgtm!

File: locales/en.js (hunk 1 above)
Comment (opets) [reply locales/en.js]: fixed, thanks

File: pages/assessment-quality-dashboard-view.js [hunk 2]
```diff
@@ -201,6 +205,13 @@ class AssessmentQualityDashboardView extends SkeletonMixin(Localizer(MobxLitElem
 				</d2l-overflow-group>
 			</div>
 		${this._renderFilters()}
+		<d2l-alert 
+			class="d2l-quiz-limit-alert"
+			?hidden=${!this.serverData._serverData?.isRecordsTruncated}
```
Comment (Vitalii-Misechko) [inline pages/assessment-quality-dashboard-view.js:210]: It's not a good idea to access private fields `_serverData`. I would suggest introducing `isRecordsTruncated` property for `serverData` object

File: pages/assessment-quality-dashboard-view.js [hunk 3]
```diff
@@ -201,6 +205,13 @@ class AssessmentQualityDashboardView extends SkeletonMixin(Localizer(MobxLitElem
 				</d2l-overflow-group>
 			</div>
 		${this._renderFilters()}
+		<d2l-alert 
+			class="d2l-quiz-limit-alert"
+			?hidden=${!this.serverData._serverData?.isRecordsTruncated} 
+			subtext=${this.localize('dashboard:quizTruncationSubtext')}
+			type="critical">
+			${this.localize('dashboard:quizTruncationTitle', { num: this.serverData.quizzes?.length?.toLocaleString() })}
```
Comment (Vitalii-Misechko) [inline pages/assessment-quality-dashboard-view.js:213]: It's odd that quizzes can be null. If that's true then we're talking about the execution of `render` function before `this.serverData` initialization. For thess cases, Lit-Element framework guide recommends property initialization in the constructor of the Lit-Element class. Could you add it?

```JS
constructor() {
  this.serverData = {
    isRecordsTruncated: false,
    quizzes: []
  }
}
```

File: pages/assessment-quality-dashboard-view.js [hunk 4]
```diff
@@ -396,7 +407,7 @@ class AssessmentQualityDashboardView extends SkeletonMixin(Localizer(MobxLitElem
 
 		return html`
 			<div class="d2l-total-quizzes">
-				${this.localize('dashboard:totalQuizzes', { num: itemCounts })}
+				${this.localize('dashboard:totalQuizzes', { num: itemCounts.toLocaleString() })}
```
Comment (Vitalii-Misechko) [inline pages/assessment-quality-dashboard-view.js:410]: nice!

File: test/pages/assessment-quality-dashboard-view.test.js [hunk 5]
```diff
@@ -149,6 +149,39 @@ describe('d2l-insights-assessment-dashboard-view', () => {
...
+					const alert = el.renderRoot.querySelector('.d2l-quiz-limit-alert');
+
+					if (isRecordsTruncated === true) {
+						expect(alert.hidden).to.be.not.true;
+						expect(alert.innerHTML.trim()).equal('Your results exceed 22 quizzes and we are unable to display all available information.');
+						expect(alert.subtext.trim()).equal('Please refine your filters or contact D2L Support for assistance.');
+					} else {
+						expect(alert.hidden).to.be.true;
+					}
+				});
```
Comment (Vitalii-Misechko) [inline test/pages/assessment-quality-dashboard-view.test.js:182]: `nit` just a minor code formatting suggestion.
1. alert.hidden can be verified once
2. your code is good by checking isRecordsTruncated for truthy value
```JS
expect(alert.hidden).to.equal(!isRecordsTruncated)
if(isRecordsTruncated) {
  expect(...
}
```

**Review by Vitalii-Misechko — COMMENTED** [review]
You have an incorrect golden visual diff screenshot

image

EDIT: don't bother with it. It seems that's acceptable by now https://d2l.slack.com/archives/CMNGJ22G0/p1631712275095500

File: pages/assessment-quality-dashboard-view.js (hunk 2 above)
Comment (opets) [reply pages/assessment-quality-dashboard-view.js]: thanks, fixed

File: pages/assessment-quality-dashboard-view.js (hunk 3 above)
Comment (opets) [reply pages/assessment-quality-dashboard-view.js]: yes, that's not needed, fixed, thanks!

File: test/pages/assessment-quality-dashboard-view.test.js (hunk 5 above)
Comment (opets) [reply test/pages/assessment-quality-dashboard-view.test.js:178]: ok, fixed

**Comment by opets** [comment]
> You have an incorrect golden visual diff screenshot
> 
> image
> 
> **EDIT**: don't bother with it. It seems that's acceptable by now https://d2l.slack.com/archives/CMNGJ22G0/p1631712275095500

yeah, sorry, it seemed to me that I did an opposite fix: committed a file with buttons 😃 
I will rewrite this file, once visualdiff generates an appropriate file

**Review by Vitalii-Misechko — APPROVED** [review]
LGTM! Thanks for the improvements!

**Comment by opets** [comment]
> > You have an incorrect golden visual diff screenshot
> > image
> > **EDIT**: don't bother with it. It seems that's acceptable by now https://d2l.slack.com/archives/CMNGJ22G0/p1631712275095500
> 
> yeah, sorry, it seemed to me that I did an opposite fix: committed a file with buttons 😃
> I will rewrite this file, once visualdiff generates an appropriate file

hm, that's strange. seems, I really added buttons in `Updating Visual Diff Goldens` but git removed them in `Merge branch 'master'`
https://github.com/Brightspace/insights-assessment-quality-dashboard/commits/a3f4d71ec5bfdacf8da857cdea24d69d6364bcde/test/visual-diff/screenshots/ci/golden/quiz-item-analysis-view-skeleton-ltr/quiz-item-analysis-view-skeleton-ltr-Desktop.png

**Pull request merged**